app/models/transaction.rb

```diff
@@ -118,6 +118,8 @@ class Transaction < ApplicationRecord
             OR listing_authors_transactions.family_name like :pattern
             OR listing_authors_transactions.display_name like :pattern)", pattern: pattern)
   }
+  scope :paid_or_confirmed, -> { where(current_state: ['paid', 'confirmed']) }
```

Comment:
`transactions` table does not have index on `current_state`, so this is a slow query. Index should be added here.